### ISSUE TITLE
help navigating an interface

### ISSUE THREAD

**Issue by jmls** [dsherret/ts-simple-ast#1213]
I have a requirement to parse a typescript interface and convert to another language. To that end, I need to obviosusly visit each definition, get the members and the type of each member

this is part of the interface defintion. There are various types, including references to other parts of the interface (`components["schemas"]["ProviderType"]` for example)
```
export interface components {
        schemas: {
            InitPayload: {
                start: number;
                level: number;
            };
            InitResponse: {
                keys: string[];
                password: string;
            };
            ProviderType: "userpass" | "oidc" | "ldap" | "saml";
            PostTune: {
                path?: string;
                data?: string[];
                description?: string;
                options?: { [key: string]: unknown };
            };
            Provider: {
                type?: components["schemas"]["ProviderType"];
                id: string;
                name: string;
                description: string;
                metadata?: { [key: string]: unknown };
                config?: { [key: string]: unknown };
            };
            Capability: {
                id: string;
                capabilities: { [key: string]: unknown };
                description?: string;
                isSecureable?: boolean;
                path: string;
                metadata: { [key: string]: unknown };
            };
        }
    }
```

I seem to be making this far more complicated than it surely needs to be but just can't wrap my head around the ts-morph api

```
const models = this.sourceFile.getInterface('components').getProperty('schemas').getChildrenOfKind(SyntaxKind.TypeLiteral)[0].getMembers()
```

seems to give me the "members" (ie InitPayload, InitResponse etc) which seem to be `PropertySignature` 

So, then in order to get the "members" of the member (ie start. level) do I need to loop through the members of the member checking the type ? 

```
        const type = definition.getType();

        switch (true) {

            case type.isNumber(): { return 'decimal' }
            case type.isAny(): { return 'JsonObject' }
            case type.isBoolean(): { return 'logical' }
            case type.isString(): { return 'character' }

            default: {
                console.log(`WARNING: unhandled type ${type.getText()}`)
            }
```

in the case of ProviderType, this is a union, right ? 

and for Provider.type I can just return "ProviderType" as the "type" as this is a referece to the above 

Basically, do I need to recursively descend through the members until I hit a base type (number, integer etc)  ?

Are there any easier ways to do this ? I've used ts-morph to get classes, methods etc and it seems easier, but for some reason I'm struggling with an interface declaration
